api/bucketsd/service.go

```diff
@@ -3218,32 +3218,37 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 			}
 		} else { // Case 2.
 			res, err := s.PowergateClient.Data.CidInfo(ctxPow, oldCid.String())
-			if err != nil {
+			isKnownToPowergate := err == nil || status.Convert(err).Code() != codes.NotFound
```

Comment:
Here we identify if Powergate is aware of the old cid by inspecting the error. I verified the Powergate client is just passing the gRPC error along here and that the Powergate server does return `codes.NotFound` in the case where the cid is unknown.

Reply:
I'm a bit confused by this line, two cases:
- if `err == nil`: ok, the easy case makes sense.
- If `err != nil && status != NotFound`:  I'm not sure this means that "is known to Powergate". If the error is "timeout" due to some network problem, then we're assuming "is known to powergate"?

Maybe I'm confused.

Thinking about what you're trying to do... I can't think of some way to answer this question.

Reply:
Maybe it would make more sense if I frame it from the other angle: is not known to powergate. That is a value with an easy to understand meaning and calculation. Change coming up.

Reply:
Ok just pushed the update. The subsequent use is a little awkward double negative reading, but I think it's worth it to have a straight forward calculation and meaning of `isNotKnownToPowergate`, no ambiguity.

Reply:
Also, just to clarify the need for this in more general terms... This is addressing the case where the `oldCid` from the latest bucket archive is not known to Powergate (because we reset Powergate at some point). This change detects the `NotFound` error, and in that case, doesn't return an error (Previously it would return any error, so the bucket would be unable to be archived). Additionally, this information must also be used to decide if we can remove a previous storage config in the case of applying a new storage config.